alphastats/gui/pages/02_Import Data.py

```diff
@@ -1,76 +1,141 @@
-    )
 
-import_data()
+st.markdown(
+    "Create a DataSet with the output of your proteomics software package and the corresponding metadata (optional). "
+)
 
-if "dataset" in st.session_state:
-    st.info("DataSet has been imported")
+# ########## Select Software
+st.markdown("##### 1. Select software and upload data")
 
-    if "distribution_plot" not in st.session_state:
-        save_plot_sampledistribution_rawdata()
+default_select_option = "<select>"
+options = [default_select_option] + list(SOFTWARE_OPTIONS.keys())
 
-    display_loaded_dataset()
+software = st.selectbox(
+    "Select your Proteomics Software",
+    options=options,
+)
+if software == default_select_option:
+    st.stop()
 
-st.markdown("### Or Load sample Dataset")
 
-if st.button("Load sample DataSet - PXD011839", key="load_sample_data"):
-    load_sample_data()
-    if "distribution_plot" not in st.session_state:
-        save_plot_sampledistribution_rawdata()
+# ########## Load Software File
 
+softwarefile = st.file_uploader(
+    SOFTWARE_OPTIONS.get(software).get("import_file"),
+    type=["csv", "tsv", "txt", "hdf"],
+)
 
-st.markdown("### To start a new session:")
+if softwarefile is None:
+    st.stop()
 
-if st.button("New Session: Import new dataset"):
-    empty_session_state()
-    st.rerun()
+softwarefile_df = load_softwarefile_df(software, softwarefile)
+
+intensity_column, index_column = show_loader_columns_selection(
+    software=software, softwarefile_df=softwarefile_df
+)
+
+loader = load_proteomics_data(
+    softwarefile_df,
+    intensity_column=intensity_column,
+    index_column=index_column,
+    software=software,
+)
+
+
+# ##########  Load Metadata File
+st.markdown("##### 3. Prepare Metadata (optional)")
+sample_column = None
+metadatafile_df = show_metadata_file_uploader(loader)
+if metadatafile_df is not None:
+    sample_column = show_select_sample_column_for_metadata(
+        metadatafile_df, software, loader
+    )
+
+
+# ##########  Create dataset
+st.markdown("##### 4. Create DataSet")
+
+dataset = None
+metadata_columns = []
+c1, c2 = st.columns(2)
+
+if c2.button("Create DataSet without metadata"):
+    dataset = DataSet(loader=loader)
+    metadata_columns = ["sample"]
+
+if c1.button("Create DataSet with metadata", disabled=metadatafile_df is None):
+    if len(metadatafile_df[sample_column].to_list()) != len(
+        metadatafile_df[sample_column].unique()
+    ):
+        raise ValueError("Sample names have to be unique.")
+
+    dataset = DataSet(
+        loader=loader,
+        metadata_path=metadatafile_df,
+        sample_column=sample_column,
+    )
+    metadata_columns = metadatafile_df.columns.to_list()
+
+if dataset is not None:
+    st.info("DataSet has been created.")
+    st.session_state["dataset"] = dataset
+    st.session_state["metadata_columns"] = metadata_columns
+    st.session_state["loader"] = loader
```

Comment:
TODO: Figure out if we even need the loader here, as the dataset has the loader as an attribute.